cognite/neat/rules/models/_rules/dms_schema.py

```diff
@@ -144,3 +165,176 @@ def validate(self) -> list[DMSSchemaError]:
                     errors.add(DuplicatedViewInDataModelError(referred_by=model.as_id(), view=view_id))
 
         return list(errors)
+
+
+@dataclass
+class PipelineSchema(DMSSchema):
+    transformations: TransformationWriteList = field(default_factory=lambda: TransformationWriteList([]))
+    databases: DatabaseWriteList = field(default_factory=lambda: DatabaseWriteList([]))
+    raw_tables: RawTableWriteList = field(default_factory=lambda: RawTableWriteList([]))
+
+    _SQL_TYPE_BY_PROPERTY_TYPE: ClassVar[dict[type[dm.PropertyType], str]] = {
+        dm.Text: "STRING",
+        dm.Int32: "INTEGER",
+        dm.Int64: "INTEGER",
+        dm.Float32: "FLOAT",
+        dm.Float64: "DOUBLE",
+        dm.Date: "DATE",
+        dm.Timestamp: "TIMESTAMP",
+        dm.TimeSeriesReference: "STRING",
+        dm.FileReference: "STRING",
+        dm.SequenceReference: "STRING",
+        dm.Boolean: "BOOLEAN",
+        dm.Json: "STRING",
+    }
+
+    @classmethod
+    def from_dms(cls, schema: DMSSchema, instance_space: str | None = None) -> "PipelineSchema":
+        if not schema.data_models:
+            raise ValueError("PipelineSchema must contain at least one data model")
+        first_data_model = schema.data_models[0]
+        # The database name is limited to 32 characters
+        database_name = first_data_model.external_id[:32]
+        instance_space = instance_space or first_data_model.space
+        database = DatabaseWrite(name=database_name)
+        parent_views = {parent for view in schema.views for parent in view.implements or []}
+        container_by_id = {container.as_id(): container for container in schema.containers}
+
+        transformations = TransformationWriteList([])
+        raw_tables = RawTableWriteList([])
```

Comment:
How do you deal with IDs of instances? Maybe I missed this, but I am not sure that I've seen you are adding `ID` column to raw tables?